pinot-segment-local/src/main/java/org/apache/pinot/segment/local/io/writer/impl/VarByteChunkForwardIndexWriterV4.java

```diff
@@ -121,6 +123,42 @@ public void putBigDecimal(BigDecimal bigDecimal) {
     putBytes(BigDecimalUtils.serialize(bigDecimal));
   }
 
+  @Override
+  public void putByteArrays(byte[][] values) {
+    // num values + length of each value
+    int size = Integer.BYTES + Integer.BYTES * values.length;
+    for (byte[] value : values) {
+      size += value.length;
+    }
+    byte[] serializedBytes = new byte[size];
+    ByteBuffer byteBuffer = ByteBuffer.wrap(serializedBytes);
+    byteBuffer.putInt(values.length);
+
+    for (byte[] value : values) {
```

Comment:
Consider using the same format as the old index `[numValues, length1, length2, ..., bytes1, bytes2, ...]`

Reply:
Ack